runtime/src/main/java/com/datastax/oss/quarkus/runtime/AbstractCassandraClientProducer.java

```diff
@@ -0,0 +1,120 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultProgrammaticDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import com.datastax.oss.quarkus.config.CassandraClientConfig;
+import com.datastax.oss.quarkus.config.CassandraClientConnectionConfig;
+import com.typesafe.config.ConfigFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.CompletionStage;
+
+public abstract class AbstractCassandraClientProducer {
+
+  private CassandraClientConfig config;
+
+  public void setCassandraClientConfig(CassandraClientConfig config) {
+    this.config = config;
+  }
+
+  private ProgrammaticDriverConfigLoaderBuilder createDriverConfigLoader() {
+    return new DefaultProgrammaticDriverConfigLoaderBuilder(
+        () -> {
+          ConfigFactory.invalidateCaches();
+          return ConfigFactory.defaultOverrides()
+              .withFallback(ConfigFactory.parseResources("application.conf"))
+              .withFallback(ConfigFactory.parseResources("application.json"))
+              .withFallback(ConfigFactory.defaultReference())
+              .resolve();
+        },
+        DefaultDriverConfigLoader.DEFAULT_ROOT_PATH) {
+      @NonNull
+      @Override
+      public DriverConfigLoader build() {
+        return new NonReloadableDriverConfigLoader(super.build());
+      }
+    };
+  }
+
+  CassandraClientConfig getCassandraClientConfig() {
+    return config;
+  }
+
+  public CqlSession createCassandraClient(CassandraClientConfig config) {
+    ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder = createDriverConfigLoader();
+    configureConnectionSettings(configLoaderBuilder, config.cassandraClientConnectionConfig);
+    CqlSessionBuilder builder = CqlSession.builder().withConfigLoader(configLoaderBuilder.build());
+    return builder.build();
+  }
+
+  private void configureConnectionSettings(
```

Comment:
Semi-nit: void methods make me cringe a bit.  Can we change this to something like withConnectionSettings() and have it return the PDCLB rather than accomplishing it's mission through side effects?

Reply:
`ProgrammaticDriverConfigLoaderBuilder` is a mutable object and has side-effects, that is why this method needs to work that way.